emma-language/src/main/scala/org/emmalanguage/compiler/lang/backend/TranslateToDataflows.scala

```diff
@@ -43,63 +49,64 @@ private[backend] trait TranslateToDataflows extends Common {
      *
      * - A tree where DataBag operations have been translated to dataflow operations.
      *
-     * @param backendSymbol The symbol of the target backend.
+     * @param backend The symbol of the target backend.
      */
-    def translateToDataflows(backendSymbol: u.ModuleSymbol): u.Tree => u.Tree = tree => {
-
-      // Figure out the order stuff
-      val (disambiguatedTree, highFuns): (u.Tree, Set[u.TermSymbol]) = Order.disambiguate(tree)
+    def translateToDataflows(backend: u.ModuleSymbol): u.Tree => u.Tree = {
+      val backendRef = Some(core.Ref(backend))
+      Order.disambiguate.andThen { case (disambiguated, highOrd) =>
+        val fetched = api.TopDown.break.withBindDefs.withBindUses
+          .accumulateWith[Set[u.TermSymbol]] {
+            case Attr.syn(core.ValDef(lhs, _), uses :: defs :: _)
+              if highOrd(lhs) => for {
+                bag <- uses.keySet diff defs.keySet
+                if api.Type.constructor(bag.info) =:= API.DataBag
+              } yield bag
+          }.traverse { // Break here!
```

Comment:
It is hard to follow the logic only from the source code, some one-liner comments in lines 57-67 will be helpful here, e.g. for line 58 something like `// accumulate bag symbols appearing in ...`.